internal/analyzers/prometheus.go

```diff
@@ -0,0 +1,271 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package analyzers
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"strings"
+
+	"github.com/prometheus-operator/poctl/internal/k8sutil"
+	v1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels" 
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func RunPrometheusAnalyzer(ctx context.Context, clientSets *k8sutil.ClientSets, name, namespace string) error {
+	prometheus, err := clientSets.MClient.MonitoringV1().Prometheuses(namespace).Get(ctx, name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return fmt.Errorf("Prometheus %s not found in namespace %s", name, namespace)
+		}
+		return fmt.Errorf("error while getting Prometheus: %v", err)
+	}
+
+	cRb, err := clientSets.KClient.RbacV1().ClusterRoleBindings().List(ctx, metav1.ListOptions{
+		LabelSelector: "prometheus=prometheus",
+	})
+	if err != nil {
+		return fmt.Errorf("failed to list RoleBindings: %w", err)
+	}
+
+	if !k8sutil.IsServiceAccountBoundToRoleBindingList(cRb, prometheus.Spec.ServiceAccountName) {
+		return fmt.Errorf("ServiceAccount %s is not bound to any RoleBindings", prometheus.Spec.ServiceAccountName)
+	}
+
+	for _, crb := range cRb.Items {
+		cr, err := clientSets.KClient.RbacV1().ClusterRoles().Get(ctx, crb.RoleRef.Name, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get ClusterRole %s", crb.RoleRef.Name)
+		}
+
+		err = checkClusterRoleRules(crb, cr)
+		if err != nil {
+			return err
+		}
+	}
+
+    validPodNS, err := checkPrometheusNamespaceSelectorsStatus(ctx, clientSets, prometheus.Spec.PodMonitorNamespaceSelector)
+    if err != nil {
+        return fmt.Errorf("PodMonitorNamespaceSelector is not properly defined: %s", err)
+    }
+
+    validProbeNS, err := checkPrometheusNamespaceSelectorsStatus(ctx, clientSets, prometheus.Spec.ProbeNamespaceSelector)
+    if err != nil {
+        return fmt.Errorf("ProbeNamespaceSelector is not properly defined: %s", err)
+    }
+
+    validServiceNS, err := checkPrometheusNamespaceSelectorsStatus(ctx, clientSets, prometheus.Spec.ServiceMonitorNamespaceSelector)
+    if err != nil {
+        return fmt.Errorf("ServiceMonitorNamespaceSelector is not properly defined: %s", err)
+    }
+
+    validScrapeNS, err := checkPrometheusNamespaceSelectorsStatus(ctx, clientSets, prometheus.Spec.ScrapeConfigNamespaceSelector)
+    if err != nil {
+        return fmt.Errorf("ScrapeConfigNamespaceSelector is not properly defined: %s", err)
+    }
+
+    if !validPodNS && !validProbeNS && !validServiceNS && !validScrapeNS {
```

Comment:
This error message is a bit unclear for the user, since it doesn't provide information about what selector has an issue.

I think we should add validation per selector right after check if the checkPrometheusNamespaceSelectorsStatus has an error

Reply:
The ideia was to log when all ns are being matched, but removed in the mean time